### ISSUE TITLE
Add unsubscribe method to Conf class

### ISSUE THREAD

**Issue by willium** [sindresorhus/conf#131]
It's great that `onDidChange` and `onDidAnyChange` return functions to destroy the watcher, but it would be great to have the listeners be internal state and then provide an `unsubscribe` method. This is especially nicer when you need to manage multiple Conf instances at once.

Before
```ts
let conf1 = undefined
let unsubscribeConf1 = undefined
let conf2 = undefined
let unsubscribeConf2 = undefined

export foo = () => {
    if (unsubscribeConf1) unsubscribeConf1()
    conf1 = new Conf(...)
    unsubscribeConf1  = conf1.onDidAnyChange(...)
}

export bar = () => {
    if (unsubscribeConf2) unsubscribeConf2()
    conf2 = new Conf(...)
    unsubscribeConf2  = conf2.onDidAnyChange(...)
}
```

After
```ts
let conf1 = undefined
let conf2 = undefined

export foo = () => {
    if (conf1) conf1.unsubscribe()
    conf1 = new Conf(...)
    conf1.onDidAnyChange(...)
}

export bar = () => {
    if (conf2) conf2.unsubscribe()
    conf2 = new Conf(...)
    conf2.onDidAnyChange(...)
}
```

**Comment by sindresorhus** [comment]
I'm not really sold on this. You would only save a few characters, but it would introduce overhead both in the implementation, the docs, and more API surface for the developer to learn. I also don't think it's common to use multiple Conf instances.

**Comment by willium** [comment]
Fair enough. I was able to achieve this fairly easily by just subclassing the conf object :)

**Comment by sindresorhus** [comment]
I wonder if we could provide something that makes React integration easier though. Maybe a React hook or something. I'm always open to ideas on how to make it simpler to use.

**Comment by willium** [comment]
I’m actually using it in electron, but inside a thread that doesn’t have access to app, so I’ve basically recreated a proxied version of electron-store. I just prefer the .current idiom from Ref over the .store one. The unsubscribe need for me is swapping config files (well really directories with the same files) based on the user. When you switch users, I want to unsubscribe from the old files and subscribe to the new ones
